src/nbl/video/CVKLogicalDevice.h

```diff
@@ -178,52 +802,382 @@ class CVKLogicalDevice final : public ILogicalDevice
+
+            vk_shaderStageCreateInfos[i].sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+            vk_shaderStageCreateInfos[i].pNext = nullptr; // pNext must be NULL or a pointer to a valid instance of VkPipelineShaderStageRequiredSubgroupSizeCreateInfoEXT
+            vk_shaderStageCreateInfos[i].flags = 0; // currently there is no way to get this in the API https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkPipelineShaderStageCreateFlagBits.html
+            vk_shaderStageCreateInfos[i].stage = static_cast<VkShaderStageFlagBits>(specShader->getStage());
+            vk_shaderStageCreateInfos[i].module = specShader->getInternalObject();
+            vk_shaderStageCreateInfos[i].pName = "main"; // Probably want to change the API of IGPUSpecializedShader to have something like getEntryPointName like theres getStage
+            vk_shaderStageCreateInfos[i].pSpecializationInfo = nullptr; // Todo(achal): Should we have a asset::ISpecializedShader::SInfo member in CVulkanSpecializedShader, otherwise I don't know how I'm gonna get the values required for VkSpecializationInfo https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkSpecializationInfo.html
+
+            vk_createInfos[i].stage = vk_shaderStageCreateInfos[i];
+
+            vk_createInfos[i].layout = VK_NULL_HANDLE;
+            if (createInfo->layout && (createInfo->layout->getAPIType() == EAT_VULKAN))
+                vk_createInfos[i].layout = static_cast<const CVulkanPipelineLayout*>(createInfo->layout.get())->getInternalObject();
+
+            vk_createInfos[i].basePipelineHandle = VK_NULL_HANDLE;
+            if (createInfo->basePipeline && (createInfo->basePipeline->getAPIType() == EAT_VULKAN))
+                vk_createInfos[i].basePipelineHandle = static_cast<const CVulkanComputePipeline*>(createInfo->basePipeline.get())->getInternalObject();
+
+            vk_createInfos[i].basePipelineIndex = createInfo->basePipelineIndex;
+        }
+        
+        VkPipeline vk_pipelines[MAX_PIPELINE_COUNT];
+        if (vkCreateComputePipelines(m_vkdev, vk_pipelineCache, static_cast<uint32_t>(createInfos.size()),
+            vk_createInfos, nullptr, vk_pipelines) == VK_SUCCESS)
+        {
+            for (size_t i = 0ull; i < createInfos.size(); ++i)
+            {
+                const auto createInfo = createInfos.begin() + i;
+
+                output[i] = core::make_smart_refctd_ptr<CVulkanComputePipeline>(
+                    core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                    core::smart_refctd_ptr(createInfo->layout),
+                    core::smart_refctd_ptr(createInfo->shader), vk_pipelines[i]);
+            }
+            return true;
+        }
+        else
+        {
+            return false;
+        }
     }
 
     core::smart_refctd_ptr<IGPURenderpassIndependentPipeline> createGPURenderpassIndependentPipeline_impl(IGPUPipelineCache* _pipelineCache,
```

Comment:
renderpass indep pipeline will be a dummy object (no vulkan object associates with it) holding parts of the state that is needed to create a real graphics pipeline

Reply:
Resolved